src/session/Manager.cc

```diff
@@ -218,9 +218,17 @@ void Manager::Insert(Session* s, bool remove_existing)
 
 void Manager::Drain()
 	{
-	for ( const auto& entry : session_map )
+	std::vector<const detail::Key*> keys;
+	keys.reserve(session_map.size());
+
+	for ( auto& entry : session_map )
+		keys.push_back(&(entry.first));
+	std::sort(keys.begin(), keys.end(), [](const detail::Key* a, const detail::Key* b) {
```

Comment:
I'm wondering if we should limit this behavior to when we running with an RNG seed (i.e., to testing). Order doesn't matter in "normal" use, and I'm a bit conconered that this could introduce noticeable (and unnecessary) delays at termination with large connection tables.

Reply:
Oh, that is a really great point - I like that.

Reply:
See the new commit.